js/apps/admin-ui/src/groups/components/GroupTree.tsx

```diff
@@ -314,14 +312,18 @@ export const GroupTree = ({
           onSelect={(_, item) => {
             if (item.id === "next") return;
             setActiveItem(item);
-            const id = item.id?.substring(item.id.lastIndexOf("/") + 1);
-            const subGroups: GroupRepresentation[] = [];
-            findGroup(groups, id!, [], subGroups);
-            setSubGroups(subGroups);
+
+            const path = findGroup(data, item.id!, []);
+            if (!subGroups.every(({ id }) => path.find((t) => t.id === id)))
+              clear();
 
             if (canViewDetails || subGroups.at(-1)?.access?.view) {
```

Comment:
Not specifically related to this fix, but when fine grained permissions are enabled, this check is not sufficient because it will not check access to the selected item and therefore will always show the `noViewRights`

Reply:
@pkeuter could you make a separate issue for that?

Reply:
Done. See https://github.com/keycloak/keycloak/pull/26230